src/clarity-angular/popover/signpost/if-open.directive.specs.ts

```diff
@@ -0,0 +1,113 @@
+// /*
+//  * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+//  * This software is released under MIT license.
+//  * The full license information can be found in LICENSE in the root directory of this project.
+//  */
+import { Component, ViewChild } from "@angular/core";
+import { IfOpenService } from "./if-open.service";
+import {TestBed} from "@angular/core/testing";
+import { IfOpenDirective } from "./if-open.directive";
+
+export default function(): void {
+    describe("IfOpen Directive", function () {
+        describe("Typescript API", function () {
+            beforeEach(function () {
+               TestBed.configureTestingModule(({
+                   declarations: [IfOpenDirective, IfOpenTest],
+                   providers: [IfOpenService]
+               }));
+               this.fixture = TestBed.createComponent(IfOpenTest);
+               this.fixture.detectChanges();
+               this.testComponent = this.fixture.componentInstance;
+               this.testElement = this.fixture.nativeElement;
+               this.clarityDirective = this.fixture.componentInstance.directive;
+               this.ifOpenService = TestBed.get(IfOpenService);
+            });
+
+            afterEach(function() {
+               this.fixture.destroy();
+            });
+
+            it("sets the open state of the directive", function() {
+                let testValue: boolean = true;
+                this.testComponent.openState = testValue;
+                this.fixture.detectChanges();
+                expect(this.clarityDirective.open).toEqual(testValue);
+            });
+
+            it("provides the current value of the open state", function() {
+                let testValue = this.clarityDirective.open;
+                expect(this.testComponent.openState).toBe(testValue);
+            });
```

Comment:
I don't understand what this test does. Your component doesn't test the output at all, and this test just checks the initial state.